operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -190,8 +191,21 @@ private String getLockName(String namespace, String name) {
 
             if (cr != null) {
                 Promise<Void> createOrUpdate = Promise.promise();
+                if (Annotations.isReconciliationPausedWithAnnotation(cr)) {
+                    S status = createStatus();
+                    Set<Condition> conditions = validate(cr);
+                    conditions.add(StatusUtils.getPausedCondition());
+                    status.setConditions(new ArrayList<>(conditions));
 
-                if (cr.getSpec() == null)   {
+                    updateStatus(reconciliation, status).onComplete(statusResult -> {
+                        if (statusResult.succeeded()) {
+                            createOrUpdate.complete();
+                        } else {
+                            createOrUpdate.fail(statusResult.cause());
+                        }
+                    });
+                    return createOrUpdate.future();
```

Comment:
Should we log (at debug) that the CR is paused? Should we have a metric for the number of paused CRs?